root.spec

```diff
@@ -22,6 +22,21 @@ Requires: freetype
 
 %define keep_archives true
 
+%define subpackageDebug true
+
+%if "%{?subpackageDebug:set}" == "set"
+%define envOptions USER_CXXFLAGS="-fdebug-prefix-map=%{cmsroot}=%{installroot} -fdebug-prefix-map=%{instroot}=%{installroot} -g"
+%endif
+# For non-linux archs disable subpackageDebug otherwise add dwz.file build dependencies.
+%if "%{?subpackageDebug:set}" == "set"
+%if %islinux
+BuildRequires: file
+%else
+%undefine subpackageDebug
+%endif # islinux
+%endif # subpackageDebug
+
+
```

Comment:
Could we get rid of multiple `%if "%{?subpackageDebug:set}" == "set"` checks? Less code, could be more readable. Also commend `add dwz.file build dependencies.` is incorrect.

I am also not sure about `-fdebug-prefix-map=%{cmsroot}=%{installroot}`. This was technically only used by ONLINE people. They always install CMSSW on specific (default?) path, which is `/opt/cmssw`. Thus this might not work for OFFLINE, where `%{installroot}` can be whatever path and it's only known at install time.

I don't even know if these flags are required. Look at https://sourceware.org/gdb/onlinedocs/gdb/Separate-Debug-Files.html for guidance of how GDB finds the debug information.

We use the old way with debug link, but all binaries with GCC 4.9.X are also marked with build ids, which can be used to locate debug files.